pkg/client/client.go

```diff
@@ -50,9 +51,25 @@ type tlsBootstrapClientImpl struct {
 	resource       string
 }
 
+var newLoadExecCredential = func() (*datamodel.ExecCredential, error) {
+	return loadExecCredential()
+}
+
+var newGetTlsConfig = func(pemCAs []byte, c *tlsBootstrapClientImpl, execCredential *datamodel.ExecCredential) (*tls.Config, error) {
+	return getTLSConfig(pemCAs, c.nextProto, execCredential.Spec.Cluster.InsecureSkipTLSVerify)
```

Comment:
I think this makes sense - I'd like to minimize usage of this pattern of using function ptrs and overriding them within tests as much as possible - _only_ to wrap calls to 3rd-party libs/pkgs